project_move_activity_tasks/wizard/project_move_activity_tasks_wizard.py

```diff
@@ -0,0 +1,122 @@
+    """Wizard Move Activity Tasks"""
+    _name = 'move.activity.tasks.wiz'
+    _description = __doc__
+
+    activity_id = fields.Many2one(
+        'project.task',
+        string='Task',
+    )
+    message = fields.Html(
+        string='Message',
+    )
+    days = fields.Integer(
+        string='Days',
+    )
+    hours = fields.Integer(
+        string='Hours',
+    )
+    minutes = fields.Integer(
+        string='Minutes',
+    )
+    moving_date = fields.Datetime()
+    child_ids = fields.One2many('project.task', 'parent_id',
+                                string="Tasks",
+                                readonly=True,
+                                compute="_compute_get_activity_childs",
+                                )
+    moving_direction = fields.Selection(string='Type',
+                                        selection=[
+                                            ('before', 'Before'),
+                                            ('after', 'After')
+                                        ],
+                                        default="after")
+    moving_type = fields.Selection(string='Type',
+                                   selection=[
+                                       ('interval', 'Interval moving'),
+                                       ('date', 'Date moving')
+                                   ],
+                                   default="interval")
+    move_activity = fields.Boolean(default=True)
+
+    def _compute_get_activity_childs(self):
+        tasks = []
+        for child in self.activity_id.child_ids:
+            if not child.is_main_task:
+                tasks.append(child.id)
+
+        self.child_ids = [(6, False, tasks)]
+
+    def _move_tasks_with_interval(self, direction, child):
+        child.write({
+            'date_start': (datetime.datetime.strptime(child.date_start,
+                                                      '%Y-%m-%d %H:%M:%S') +
+                           direction * datetime.timedelta(
+                days=self.days,
+                hours=self.hours,
+                minutes=self.minutes)).strftime('%Y-%m-%d %H:%M:%S'),
+            'date_end': (datetime.datetime.strptime(child.date_end,
+                                                    '%Y-%m-%d %H:%M:%S') +
+                         direction * datetime.timedelta(
+                days=self.days,
+                hours=self.hours,
+                minutes=self.minutes)).strftime('%Y-%m-%d %H:%M:%S')
+        })
+
+    def _move_tasks_to_date(self, child):
+        moving_date = datetime.datetime.strptime(self.moving_date,
+                                                 '%Y-%m-%d %H:%M:%S')
+        date_start = datetime.datetime.strptime(child.date_start,
+                                                '%Y-%m-%d %H:%M:%S')
+        date_end = datetime.datetime.strptime(child.date_end,
+                                              '%Y-%m-%d %H:%M:%S')
+        child.write({
```

Comment:
So you moving the activity and its children, all to same date? The requirement wasn't that with the new date, a delta was calculated (old date of the activity - new date) and apply that delta to its children

Reply:
I should validate the requirements. I think that the first functionality of interval could be used for this purpose or the user can move just the Activity/Main task by checking the related boolean and move the rest of the tasks separately in a second operation.